### DISCUSSION TITLE
Is there a way to provide the persistent object to the map method ?

### DISCUSSION THREAD

**Discussion by MilanRaval** [leonardoporro/Detached-Mapper#88] [Q&A]
Is there a way to provide the persistent object instance to the map method as a parameter ? And use the Detached Mapper for only comparing and setting entity state ?

**Comment by leonardoporro** [comment]
Sorry, I missed this post! don't know why... 

Actually you can write your own map method and put this inside of it: 

``` csharp
            var profileKey = ProfileKey.Empty; // default profile.

            var entity = new { Id = 1 }; // your tracked entity
            var dto = new { Id = 1 }; // some dto, which could be another instace of entity
            var mapper = dbContext.GetEntityMapper(); // get the EntityMapper instance associated to the dbContext.
            var parameters = new MapParameters(); // create parameters, same as the ones supplied in regular Map
            // create a context. this is the link between the mapper and EntityFramework, and it helps to track entity states and apply some behaviors
            var context = new EntityMapContext(mapper.Options.ConcurrencyTokens, dbContext, mapper.GetQueryProvider(profileKey), parameters);

            // now get the mapper. EntityMapper is a kind of adapter/façade of Mapper, there is 1 mapper per profile
            var result = mapper.GetMapper(profileKey).Map(dto, dto.GetType(), entity, entity.GetType(), context);
```  

I wouldn't like to add this as a feature, because the QueryProvider should take care of loading the entities and adding the right Includes. etc.
If I make a method like this, public, it will make my life harder trying to give support to the libray :) (I'm already overwelmed)

**Reply by MilanRaval** [reply]
thank you !

**Comment by leonardoporro** [comment]
In 7.2 there was a large refactor about this. 
Will try to add an overload that takes the target entity as a parameter..

**Reply by cbordeman** [reply]
Trying to adapt this to the latest code, and to start ConcurrencyTokens doesn't exist.  Is there a way to do this in 7.2?

**Reply by leonardoporro** [reply]
Using version 8+ you can do:
`_context.GetEntityMapper().Map(_context, [your entity], null);`

If you want to take a look at the 
[code](https://github.com/leonardoporro/Detached-Mapper/blob/main/src/Detached.Mappers.EntityFramework/EntityMapper.cs),
basically, you may omit the "loader" part now.

For this to work, make sure that [your entity] is loaded and attached to the dbcontext. 
Passing your own version of the tree, without using the Loader may generate side effects that I didn't know about yet.

Version 8 should work with .net 6 to 8, but it is a fresh change so please let me know if you try it and find anything
